group-coordinator/src/main/java/org/apache/kafka/coordinator/group/metrics/GroupCoordinatorMetrics.java

```diff
@@ -281,100 +277,99 @@ public void onUpdateLastCommittedOffset(TopicPartition tp, long offset) {
-            @Override
-            public Long value() {
-                return numGenericGroupsCompletingRebalanceCount();
-            }
-        });
-
-        registry.newGauge(NUM_GENERIC_GROUPS_STABLE, new Gauge<Long>() {
-            @Override
-            public Long value() {
-                return numGenericGroupsStableCount();
-            }
-        });
-
-        registry.newGauge(NUM_GENERIC_GROUPS_DEAD, new Gauge<Long>() {
-            @Override
-            public Long value() {
-                return numGenericGroupsDeadCount();
-            }
-        });
-
-        registry.newGauge(NUM_GENERIC_GROUPS_EMPTY, new Gauge<Long>() {
-            @Override
-            public Long value() {
-                return numGenericGroupsEmptyCount();
-            }
-        });
-
-        registry.newGauge(NUM_CONSUMER_GROUPS, new Gauge<Long>() {
-            @Override
-            public Long value() {
-                return numConsumerGroups();
-            }
-        });
-
-        registry.newGauge(NUM_CONSUMER_GROUPS_EMPTY, new Gauge<Long>() {
+        registry.newGauge(NUM_GENERIC_GROUPS_PREPARING_REBALANCE, new com.yammer.metrics.core.Gauge<Long>() {
             @Override
             public Long value() {
-                return numConsumerGroupsEmpty();
+                return numGenericGroupsPreparingRebalance();
             }
         });
 
-        registry.newGauge(NUM_CONSUMER_GROUPS_ASSIGNING, new Gauge<Long>() {
+        registry.newGauge(NUM_GENERIC_GROUPS_COMPLETING_REBALANCE, new com.yammer.metrics.core.Gauge<Long>() {
             @Override
             public Long value() {
-                return numConsumerGroupsAssigning();
+                return numGenericGroupsCompletingRebalance();
             }
         });
 
-        registry.newGauge(NUM_CONSUMER_GROUPS_RECONCILING, new Gauge<Long>() {
+        registry.newGauge(NUM_GENERIC_GROUPS_STABLE, new com.yammer.metrics.core.Gauge<Long>() {
             @Override
             public Long value() {
-                return numConsumerGroupsReconciling();
+                return numGenericGroupsStable();
             }
         });
 
-        registry.newGauge(NUM_CONSUMER_GROUPS_STABLE, new Gauge<Long>() {
+        registry.newGauge(NUM_GENERIC_GROUPS_DEAD, new com.yammer.metrics.core.Gauge<Long>() {
             @Override
             public Long value() {
-                return numConsumerGroupsStable();
+                return numGenericGroupsDead();
             }
         });
 
-        registry.newGauge(NUM_CONSUMER_GROUPS_DEAD, new Gauge<Long>() {
+        registry.newGauge(NUM_GENERIC_GROUPS_EMPTY, new com.yammer.metrics.core.Gauge<Long>() {
             @Override
             public Long value() {
-                return numConsumerGroupsDead();
+                return numGenericGroupsEmpty();
             }
         });
+        metrics.addMetric(
+            metrics.metricName(GROUPS_COUNT_METRIC_NAME, METRICS_GROUP, Collections.singletonMap(
```

Comment:
I wonder if we could add the description to all the new metrics. Have you considered it?